### DISCUSSION TITLE
Script for steering an array antenna beam in linux

### DISCUSSION THREAD

**Discussion by erikajob91** [ansys/pyaedt#5191] [Q&A] [answered]
In linux GUI I'm not able to execute the script for steering an antenna array beam (see attached file
[Finite_Array_Beam_Angle.zip](https://github.com/user-attachments/files/17077589/Finite_Array_Beam_Angle.zip)
). Is it possible to write a new version using pyAEDT apis ?

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @erikajob91, yes it is possible. 

But before you start any refactoring, I encourage you to take a look to the Farfield visualization class, which is much more powerful.

https://aedt.docs.pyansys.com/version/stable/API/Visualization.html#farfield

This class allows you to extract the farfield data, and with python perform all farfield postprocessing, like the beam steering, active s parameters...

**Reply by erikajob91** [reply]
Hi,
is it possible to have an example as starting point ?

**Comment by Samuelopez-ansys** [comment]
Sure! There you are:
https://aedt.docs.pyansys.com/version/stable/examples/02-HFSS/Array.html#load-far-field-data
